src/Microsoft.ML.Data/Transforms/RowShufflingTransformer.cs

```diff
@@ -95,7 +97,8 @@ private static VersionInfo GetVersionInfo()
         /// <param name="poolRows">The pool will have this many rows</param>
         /// <param name="poolOnly">If true, the transform will not attempt to shuffle the input cursor but only shuffle based on the pool. This parameter has no effect if the input data was not itself shufflable.</param>
         /// <param name="forceShuffle">If true, the transform will always provide a shuffled view.</param>
-        public RowShufflingTransformer(IHostEnvironment env,
+        [BestFriend]
+        internal RowShufflingTransformer(IHostEnvironment env,
```

Comment:
>internal [](start = 8, length = 8)

Similar notes. there is no point in internalizing these.

Reply:
Note that we had a drive to internalize constructors of `ITransformer` implementations, not `IDataTransform` implementations. (The goal there is to just internalize them altogether, as we see in this case has already been done.)

---
In reply to: [254349247](https://github.com/dotnet/machinelearning/pull/2427#discussion_r254349247) [](ancestors = 254349247)